src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -62,6 +67,35 @@ protected async Task<IActionResult> GetDetector(TResource resource, string detec
             return detectorResponse == null ? (IActionResult)NotFound() : Ok(DiagnosticApiResponse.FromCsxResponse(detectorResponse.Item1, detectorResponse.Item2));
         }
 
+        protected async Task<IActionResult> ListGists()
+        {
+            await _sourceWatcherService.Watcher.WaitForFirstCompletion();
+
+            return Ok(_gistCache.GetAll()
+                .Select(p => new Dictionary<string, string>
+                {
+                    { "Id", p.Id },
```

Comment:
would recommend adding a GistDefinition Model in the Diagnostics.Model class and use that here as well as in Authoring Experience.